tests/features/composite-keys/custom-pivot-entity-with-field-name-and-discriminator.test.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Collection,
+  Entity,
+  EntityCaseNamingStrategy,
+  ManyToMany,
+  ManyToOne,
+  MikroORM,
+  PrimaryKey,
+  Property,
+} from '@mikro-orm/core';
+
+@Entity({ abstract: true })
+export abstract class AbstractEntity {
+
+  @PrimaryKey()
+  id!: number;
+
+}
+
+@Entity({
+  discriminatorColumn: 'role',
+  discriminatorMap: {
+    CREATOR: 'Creator',
+  },
+})
+class User extends AbstractEntity {
+
+  @Property({ default: 'CREATOR' })
+  role: 'CREATOR' = 'CREATOR' as const;
+
+}
+
+@Entity()
+class Creator extends User {
+
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+  @ManyToMany({ entity: () => Task, pivotEntity: () => CreatorsOnTasks })
+  tasks = new Collection<Task>(this);
+
+}
+
+@Entity({ tableName: 'Task' })
+class Task extends AbstractEntity {
+
+  @ManyToMany(() => Creator, c => c.tasks)
+  creators = new Collection<Creator>(this);
+
+}
+
+@Entity({ tableName: 'CreatorsOnTasks' })
+class CreatorsOnTasks {
+
+  @ManyToOne({ primary: true, fieldName: 'creatorId', entity: () => Creator })
+  creator!: Creator;
+
+  @ManyToMany({ primary: true, fieldName: 'taskId', entity: () => Task })
```

Comment:
we should validate this, only m:1 and 1:1 owners can be a primary key

```suggestion
  @ManyToOne({ primary: true, fieldName: 'taskId', entity: () => Task })
```